util/network-devp2p/src/ip_utils.rs

```diff
@@ -358,6 +362,81 @@ pub fn map_external_address(local: &NodeEndpoint) -> Option<NodeEndpoint> {
 	None
 }
 
+fn search_natpmp(local: &NodeEndpoint) -> Option<NodeEndpoint> {
+	if let SocketAddr::V4(ref local_addr) = local.address {
+		let local_port = local_addr.port();
+		let local_udp_port = local.udp_port;
+
+		let search_gateway_child = ::std::thread::spawn(move || {
+			let get_public_addr = |n: &mut Natpmp| {
+				n.send_public_address_request()?;
+				::std::thread::sleep(Duration::from_millis(50));
```

Comment:
Use a (documented) `const` for the delay, with a comment explaining why its value was picked.

Reply:
done